securesystemslib/metadata.py

```diff
@@ -97,3 +99,79 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+
+class Parser:
```

Comment:
I developed this type of Generic Payload Parser which can be derived according to tuf/in-toto requirements.
This parser has two methods, `check_type` to check `payload_type` and `parse` to parse the `payload`.
Similarly a `serialize` method can be added to this parser which generate the Envelope by serializing the payload.